src/lib.rs

```diff
@@ -225,74 +225,89 @@ impl<Message> MessageFilter<Message> where Message: PartialOrd + Ord + Clone + :
 
 #[cfg(test)]
 mod test {
-    fn generate_random_vec<T>(len: usize) -> Vec<T>
-        where T: ::rand::Rand {
-        let mut vec = Vec::<T>::with_capacity(len);
-        for _ in 0..len {
-            vec.push(::rand::random::<T>());
-        }
-        vec
-    }
-
     #[test]
     fn size_only() {
-        let size = 10usize;
+        let size = ::rand::random::<u8>() as usize;
         let mut msg_filter = super::MessageFilter::<usize>::with_capacity(size);
+        assert_eq!(::time::Duration::max_value(), msg_filter.time_to_live);
+        assert_eq!(size, msg_filter.capacity);
 
-        for i in 0..10 {
-            println!("i : {} ", i);
+        // Add `size` messages - all should be added.
+        for i in 0..size {
             assert_eq!(msg_filter.len(), i);
             msg_filter.add(i);
             assert_eq!(msg_filter.len(), i + 1);
         }
 
-        for i in 10..1000 {
+        // Check all added messages remain.
+        assert!((0..size).all(|index| msg_filter.check(&index)));
+
+        // Add further messages - all should be added, each time pushing out the oldest message.
+        for i in size..1000 {
```

Comment:
do you intend not to run the following test in case if size is >= 1000 ? If you do want to run then you might be interested in doing:
```
const UPPER_DELTA: usize = 1000;
....
if size >= ::std::usize::MAX - UPPER_DELTA {
    size = ::std::usize::MAX - UPPER_DELTA;
}
....
for i in size..(size + UPPER_DELTA) {
    ...
}
```